File: Leetcode/easy/13.swift
class Solution {
    func romanToInt(_ s: String) -> Int {
        let numerals: [Character: Int] = [
                                          "I": 1,
                                          "V": 5,
                                          "X": 10,
                                          "L": 50,
                                          "C": 100,
                                          "D": 500,
                                          "M": 1000
                                         ]
        
        let arr = Array(s)
        var num = 0
        var i = 0
        
        while i < arr.count - 1 {
            let value = numerals[arr[i]]!
            let next = numerals[arr[i+1]]!
            
            num += (value < next) ? -value : value
            
            i += 1
        }
        
        num += numerals[arr.last!]!
        
        return num
    }
}

File: Leetcode/easy/13.cpp
class Solution {
public:
    int romanToInt(string s) {
        map<char, int> m;
        m.insert(make_pair('I', 1));
        m.insert(make_pair('V', 5));
        m.insert(make_pair('X', 10));
        m.insert(make_pair('L', 50));
        m.insert(make_pair('C', 100));
        m.insert(make_pair('D', 500));
        m.insert(make_pair('M', 1000));
        
        int sum = 0;
        int num = 0;
        int length = s.length();
        for (int i = 0; i < length ;) {
            if ( i == (length - 1) || m[s[i]] >= m[s[i+1]]) {
                num = m[s[i]];
                ++i;
            } else {
                num = m[s[i+1]] - m[s[i]];
                i += 2;
            }
            
            sum += num;
        }
        
        return sum;
    }
};